### ISSUE TITLE
Editor shows 124 Failed Jobs when started for the first time on a project

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#14692]
**Describe the bug**
When a project is launched for the first time it displays 124 Failed Jobs even though there are none in the Asset Processor. Restarting the Editor fixes this issue.

This was also observed on Linux and MacOS.

**Steps to reproduce**
1. Create and build a Default Template project.
2. Launch the project.
3. Look at the bottom right corner (Pending Jobs/Failed Jobs) until Asset Processor is idle.

**Expected behavior**
No Failed Jobs are counted.

**Actual behavior**
There are 124 Failed Jobs listed in the Editor, however there aren't any in the Asset Processor. Restarting the Editor displays 0 properly.

**Video**

https://user-images.githubusercontent.com/86952082/220332142-bd8a0de1-f121-40ae-8959-0a49896aecb1.mp4

**Found in Branch**
Development (fde60c8)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
fde60c8bb024abedee52abdcf07244b541c70b29

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by AMZN-stankowi** [comment]
I think this is a side effect of how the Editor shows failures, this is intended behavior from the original design, and not a bug. We should revisit the design, though. It shows if any failure happens, and not just the current failure count. So if a failure occurs but it's fixed before Asset Processor closes, you'll see zero failures in Asset Processor and a failure in the editor.

You did have asset processing failures, but they were resolved before asset processor went idle, because I'm specifically working on that case. Right now, if you start processing from an empty asset cache, FBX files will usually process before a source dependency is available. Those FBX files will then fail to process. However, later in asset processing, the source dependencies become available as intermediate assets. Because the dependency becomes available, the FBX files reprocess, and this time they succeed.

So, in short: Not a bug, this is how this was originally designed, but I think we should keep the ticket open and change the design.
